client/src/controllers/ProgressController.ts

```diff
@@ -0,0 +1,138 @@
+import { Application, Controller } from '@hotwired/stimulus';
+import { gettext } from '../utils/gettext';
+
+/**
+ * <button
+ *   type="submit"
+ *   class="button button-longrunning"
+ *   data-controller="w-progress"
+ *   data-w-progress-active-class="button-longrunning-active"
+ *   data-w-progress-active-value="{% trans 'Signing in…' %}"
+ *   data-w-progress-duration-seconds-value="40"
+ *   data-action="w-progress#activate"
+ * >
+ *  {% icon name="spinner" %}
+ *  <em data-w-progress-target="label">{% trans 'Sign in' %}</em>
+ * </button>
+ */
+export class ProgressController extends Controller {
+  static classes = ['active'];
+  static targets = ['label'];
+  static values = {
+    active: { default: gettext('Loading'), type: String },
```

Comment:
```suggestion
    active: { default: '', type: String },
```

I think the actual existing default behaviour is to not set any active label, only do it if 'clicked-text' is set, so we should honour this existing behaviour and not have a default here.